src/components/project/project.service.ts

```diff
@@ -499,7 +515,9 @@ export class ProjectService {
     return {
       ...parseBaseNodeProperties(result.node),
       ...securedProps,
-      sensitivity: props.sensitivity,
+      sensitivity:
+        getHighestSensitivity(result.languageSensitivityList) ||
+        props.sensitivity,
```

Comment:
Falling back to a database value doesn't make sense. Would translation projects even have this property?
Since we default to High we can just return that instead of undefined in your getHighest function.

Reply:
Yes, both projects will have this property because both create a **High** sensitivity property on create, just that internship projects will have this node changed on update.

The getHighest function is just if given a sensitivity list and return the highest, I'd like to keep this modularized.

Reply:
Ok can we just fallback to High here then? 
```tsx
sensitivity: getHighestSensitivity(result.languageSensitivityList) ?? Sensitivity.High,
```

I think it's confusing that it is conditionally calculated from a different stored value when the logic really is just defaulting to high.

Reply:
Oh I see this is for both types of projects. You're just banking on the fact that an internship project will have an empty language list.

Reply:
exactly

Reply:
Can you add a comment stating this? This isn't immediately apparent. Also that translation projects get a default of high in db on create, which is only used when there is no engagements.

Reply:
added comment, see if it makes sense

Reply:
Beautiful, thanks Frank